### PR TITLE
Chore: Update grafana-plugin-sdk

### PR DESCRIPTION
**Pull request by andresmgot**
**What is this feature?**

Upgrade grafana-plugin-sdk-go to v0.215.0. Adapt to the breaking change for HTTP headers.

### CODE REVIEW

File: pkg/services/datasources/service/datasource.go [hunk 1]
```diff
@@ -680,7 +680,7 @@ func (s *Service) getCustomHeaders(jsonData *simplejson.Json, decryptedValues ma
 		}
 
 		if val, ok := decryptedValues[headerValueSuffix]; ok {
-			headers[key] = val
+			headers.Add(key, val)
```
Comment (wbrowne) [inline pkg/services/datasources/service/datasource.go:683]: Are we sure we want to `Add` here and not `Set`?

**Review by wbrowne — APPROVED** [review]
Looks good, just one question!

File: pkg/services/datasources/service/datasource.go (hunk 1 above)
Comment (andresmgot) [reply pkg/services/datasources/service/datasource.go:683]: Yes, if not, only the last value set in the SecureJSONData will be set.

Comment (sathieu) [reply pkg/services/datasources/service/datasource.go:683]: @andresmgot This is the goal of this: allow to set the same header several time. See https://github.com/grafana/grafana-plugin-sdk-go/issues/829

Comment (andresmgot) [reply pkg/services/datasources/service/datasource.go:683]: That's why we need to use `.Add`, not `.Set`, right? Unless I am missing something.

Comment (wbrowne) [reply pkg/services/datasources/service/datasource.go:683]: Thought so - would be good to add a test here in that case.

**Review by marefr — APPROVED** [review]
LGTM

File: pkg/services/datasources/service/datasource.go (hunk 1 above)
Comment (andresmgot) [reply pkg/services/datasources/service/datasource.go:683]: :+1: added a test to cover this

**Review by aangelisc — APPROVED** [review]
Azure changes LGTM 😊

**Review by wbrowne — APPROVED** [review]
🚀

**Pull request merged**